### PR TITLE
Add side by side script to upload to bintray.

### PR DESCRIPTION
**Pull request by sjudd**
This allows us to test uploading to bintray without eliminating our
ability to upload to sonatype and maven central directly. If uploading
to bintray seems to work for a version or two and we figure out syncing
to maven central and snapshots, we can later remove the
maven_central.gradle script entirely.

Progress towards #2304 / #544.

At the moment the big piece that seems to be missing is some way to upload snapshots from travis. It sort of seems like jfrog/bintray have something, I just haven't figured out how to do it yet.

### CODE REVIEW

File: library/build.gradle [hunk 1]
```diff
@@ -143,14 +143,3 @@ afterEvaluate {
...
 apply from: "${rootProject.projectDir}/scripts/upload.gradle"
-// exclude <dependency> tag for android support-v4 library from :glide's pom
-// this will ensure that this warning will not prevent the build from completing:
-// Module 'com.github.bumptech.glide:glide:4.0.0-SNAPSHOT' depends on one or more Android Libraries
-// but is a jar. Most users will need to override support-v4 version anyway if a newer version is
-// available
-// TODO make support-v4 a <scope>runtime</scope> dependency in pom.xml
-afterEvaluate {
-    uploadArchives.repositories.mavenDeployer.pom.whenConfigured { p ->
-        p.dependencies = p.dependencies.findAll { dep -> dep.artifactId != "support-v4" }
```
Comment (TWiStErRob) [inline library/build.gradle]: Where did this go?

File: scripts/artifacts.gradle [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    if (isAndroidProject) {
+        def releaseVariants = project.android.libraryVariants.findAll {
+            it.buildType.name.equalsIgnoreCase('release')
+        }
+
+        def getAndroidCompileSdkVersion = project.android.compileSdkVersion
+
+        def getAndroidSdkDirectory = project.android.sdkDirectory
+
+        def getAndroidJar = "${getAndroidSdkDirectory}/platforms/${getAndroidCompileSdkVersion}/android.jar"
```
Comment (TWiStErRob) [inline scripts/artifacts.gradle]: note: `glide/build.gradle` may duplicate some this code

File: scripts/upload.gradle [hunk 3]
```diff
@@ -1,198 +1,7 @@
-/*
```
Comment (TWiStErRob) [inline scripts/upload.gradle]: hmm, this split may conflict with #2426, I'll resolve the changes requested there this week!

File: library/build.gradle (hunk 1 above)
Comment (sjudd) [reply library/build.gradle]: Just deleted, we don't depend on v4 and we already use provided for our support library dependencies.

File: scripts/upload.gradle (hunk 3 above)
Comment (sjudd) [reply scripts/upload.gradle]: Resolved I think.

Comment (TWiStErRob) [reply scripts/upload.gradle]: Looks good.

**Pull request closed without merging**